### PR TITLE
Replace custom reloadable Key/TrustManager

### PR DESCRIPTION
**Pull request by jkakavas**
In JVMs running in FIPS approved mode, only SunJSSE TrustManagers
and KeyManagers can be used. This commit replaces all the custom
KeyManagers and TrustManagers (ReloadableKeyManager,
ReloadableTrustManager, EmptyKeyManager, EmptyTrustManager) with
instances of X509ExtendedKeyManager and X509ExtendedTrustManager.
Reloadability is now ensured by a volatile instance of SSLContext
in SSLContectHolder.
SSLConfigurationReloaderTests use the reloadable SSLContext to
initialize HTTP Clients and Servers and use these for testing the
key material and trust relations.

### CODE REVIEW

**Comment by jkakavas** [comment]
CI fails because of https://github.com/elastic/elasticsearch/issues/30507, will rerun tests once it is resolved.

**Comment by jtibshirani** [comment]
@jkakavas I just disabled`SharedClusterSnapshotRestoreIT#testSnapshotSucceedsAfterSnapshotFailure`, so I think you should be able to re-run tests: https://github.com/elastic/elasticsearch/commit/1112fac206732cb97483ee44a280cad9f2e8612e

**Comment by jkakavas** [comment]
run gradle build tests

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLService.java [hunk 1]
```diff
@@ -560,258 +553,70 @@ private static SSLSocket createWithPermissions(CheckedSupplier<Socket, IOExcepti
...
+        private void reloadSslContext() {
+            try {
+                X509ExtendedKeyManager loadedKeyManager = Optional.ofNullable(keyConfig.createKeyManager(env)).
+                        orElse(getEmptyKeyManager());
+                X509ExtendedTrustManager loadedTrustManager = Optional.ofNullable(trustConfig.createTrustManager(env)).
+                        orElse(getEmptyTrustManager());
+                SSLContext loadedSslContext = SSLContext.getInstance(sslContextAlgorithm(sslConfiguration.supportedProtocols()));
+                loadedSslContext.init(new X509ExtendedKeyManager[]{loadedKeyManager},
+                    new X509ExtendedTrustManager[]{loadedTrustManager}, null);
+                supportedCiphers(loadedSslContext.getSupportedSSLParameters().getCipherSuites(), sslConfiguration.cipherSuites(), true);
```
Comment (jaymode) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLService.java]: I do not think we should log here. This is on the reload of a file and not an update to the ciphers settings

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ssl/SSLConfigurationReloaderTests.java [hunk 2]
```diff
@@ -86,127 +96,127 @@ public void testReloadingKeyStore() throws Exception {
...
-
-        final Runnable modifier = () -> {
-            try {
-                // modify it
-                KeyStore keyStore = KeyStore.getInstance("jks");
-                keyStore.load(null, null);
-                final KeyPair keyPair = CertUtils.generateKeyPair(512);
-                X509Certificate cert = CertUtils.generateSignedCertificate(new X500Principal("CN=testReloadingKeyStore"), null, keyPair,
+        //Load HTTPClient only once. Client uses the same store as a truststore
+        try(CloseableHttpClient client = getSSLClient(keystorePath, "testnode")) {
```
Comment (jaymode) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ssl/SSLConfigurationReloaderTests.java]: space between `try` and `(`

**Review by jaymode — APPROVED** [review]
I left some minor comments. Otherwise LGTM

**Comment by jkakavas** [comment]
run gradle build tests

**Review by tvernum — APPROVED** [review]
LGTM

**Pull request merged**